### PR TITLE
ChangeLog addition: SR-7139: Support enforcement of exclusive memory

### PR DESCRIPTION
**Pull request by atrick**
…ase builds.

### CODE REVIEW

**Comment by atrick** [comment]
@swift-ci smoke test.

**Comment by benrimmington** [comment]
@atrick You'll also need to add a URL to the end of this file.

```diff
 [SR-4248]: <https://bugs.swift.org/browse/SR-4248>
+[SR-7139]: <https://bugs.swift.org/browse/SR-7139>
 [SR-7251]: <https://bugs.swift.org/browse/SR-7251>
```

File: CHANGELOG.md [hunk 1]
```diff
@@ -24,6 +24,20 @@ CHANGELOG
...
+* [SR-7139][]:
+
+  Exclusive memory access is now enforced at run-time by default in
+  optimized (-O/-Osize) builds. Programs that violate exclusivity will
+  trap at run-time with an "overlapping access" diagnostic
+  message. This can be disabled via a command line flag:
+  -enforce-exclusivity=unchecked, but doing do may result in undefined
+  behavior.
+
+  Run-time violations of exclusivity typically result from
```
Comment (krilnon) [inline CHANGELOG.md]:
```suggestion
  Runtime violations of exclusivity typically result from
```

**Review by krilnon — COMMENTED** [review]
Some style changes could help here.

File: CHANGELOG.md [hunk 2]
```diff
@@ -24,6 +24,20 @@ CHANGELOG
 Swift 5.0
 ---------
 
+* [SR-7139][]:
+
+  Exclusive memory access is now enforced at runtime by default in
+  optimized (`-O`/`-Osize`) builds. Programs that violate exclusivity will
+  trap at runtime with an "overlapping access" diagnostic
+  message. This can be disabled via a command line flag:
+  `-enforce-exclusivity=unchecked`, but doing do may result in undefined
```
Comment (benrimmington) [inline CHANGELOG.md]:
```suggestion
  `-enforce-exclusivity=unchecked`, but doing so may result in undefined
```

**Pull request merged**